### PR TITLE
handle RFC 82 entity tags in entity JSON parsing, including schema-based

### PR DESCRIPTION
**Pull request by cdisselkoen**
## Description of changes

Handles entity tags (RFC 82) in entity JSON parsing, including schema-based entity JSON parsing.  

## Issue #, if available

[RFC 82](https://github.com/cedar-policy/rfcs/blob/emina/entity-tags/text/0082-entity-tags.md)

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).

I confirm that this PR (choose one, and delete the other options):

Changelog update added with a previous PR.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)

### CODE REVIEW

**Review by khieta — APPROVED** [review]
Lgtm, pending resolution of CI failures.

**Comment by khieta** [comment]
The downstream `cedar-drt` and `cedar-java` failures are ok to ignore -- the `cedar-drt`one  will presumably be resolved by another PR, and the `cedar-java` one is an issue not caused by this PR.

But the `cedar-wasm` failures should be resolved before merging.

**Review by shaobo-he-aws — APPROVED** [review]

**Pull request merged**
